python/src/scipp/utils/get.py

```diff
@@ -0,0 +1,30 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+# @author Gregory Tucker
+
+def get(x, key, default=None):
+    """
+    Get the value associated with the provided key from a `dict`-like object
+
+    :param x: dictlike
+        Any object with an `items()` *or* both `keys()` and `values()` method(s)
+    :param key:
+        The key to extract from the provided `dict`-like object`
+    :param default:
+        A default value to return if the provided key is not present
+    :return:
+        Either the value associated with the key or the default if the key is missing
+        or the object does not implement the required methods.
+    """
+
+    return_value = default
+    if hasattr(x, 'items') and callable(x.items):
+        for k, v in x.items():
+            if k == key:
+                return_value = v
+    elif hasattr(x, 'keys') and callable(x.keys) and hasattr(x, 'values') and callable(x.values):
+        for k, v in zip(x.keys(), x.values()):
+            if k == key:
+                return_value = v
+    return return_value
```

Comment:
Why not simply
```suggestion
    return x[key] if key in x else default
```

Reply:
I think it's fair to blame this on my lack of knowledge of how the `in` operator works.

In my contrived test cases, with objects defining `items` or `keys` and `values` methods but *not* `__contains__` or `__iter__`, your proposal raises an error. But since the tests in `python/tests/utils/get_test.py` were only added to test my overwrought solution *and* the necessary methods exist in the classes which will use this `get` I think your suggestion is the right choice.